tests/test_run_app.py

```diff
@@ -685,6 +686,24 @@ async def on_startup(app):
     exc_handler.assert_called_with(patched_loop, msg)
 
 
+def test_run_app_keepalive_timeout(patched_loop):
+    new_timeout = 1234
+    base_runner_init_mock = mock.MagicMock()
+    base_runner_init_orig = BaseRunner.__init__
+
+    def base_runner_init_spy(self, *args, **kwargs):
+        base_runner_init_mock(*args, **kwargs)
+        base_runner_init_orig(self, *args, **kwargs)
```

Comment:
Have you tried this?
```suggestion
        super().__init__(*args, **kwargs)
```

Or maybe apply a decorator approach.

Reply:
Just to be clear, do you mean writing a custom decorator to encapsulate those ~three variables?